src/RCDEF.CPP

```diff
@@ -609,27 +617,31 @@ int CDEC main( SI argc, char * argv[] )
 	/* check include directory argument if specified */
 	if (HFILESOUT)              /* if argv[6] not NUL */
 	{
-		getcwd( dbuff, 80);
+		filesys::path dbuff = filesys::current_path();
 		incdir = argv[6];
-		if (chdir(incdir))
+		std::error_code ec;
+		filesys::current_path(filesys::path(incdir), ec);
+		if (ec)
 		{
-			printf( "\nCannot find include directory '%s'.\n", incdir);
+			printf( "\n\"%s\": %s", incdir, ec.message().c_str());
 			Errcount++;
 		}
-		chdir(dbuff);
+		filesys::current_path(dbuff);
 	}
 
 	/* check cpp file directory argument if specified */
 	if (CFILESOUT)              /* if argv[10] not NUL */
 	{
-		getcwd( dbuff, 80);
+		filesys::path dbuff = filesys::current_path();
 		cFilesDir = argv[10];
-		if (chdir(cFilesDir))
+		std::error_code ec;
+		filesys::current_path(filesys::path(cFilesDir), ec);
+		if (ec)
 		{
-			printf( "\nCannot find include directory '%s'.\n", cFilesDir);
+			printf("\n\"%s\": %s", cFilesDir, ec.message().c_str());
```

Comment:
Did you test this type of error?  Does the message provide appropriate info?  The old message said "include directory", so the user had some clue about what the problem is.

Reply:
I'm not sure that is the only possible error you can encounter from changing directories. If the path doesn't exist it will say something like: `"path/to/dir": Path not found`. Another possibility is that the directory doesn't have read access (maybe?). Maybe we can provide more context, like "ec.message() for cpp file directory argument"